solr/core/src/java/org/apache/solr/search/SolrIndexSearcher.java

```diff
@@ -775,7 +792,7 @@ public QueryResult search(QueryResult qr, QueryCommand cmd) throws IOException {
   protected void search(List<LeafReaderContext> leaves, Weight weight, Collector collector)
       throws IOException {
     QueryLimits queryLimits = QueryLimits.getCurrentLimits();
-    if (useExitableDirectoryReader || !queryLimits.isLimitsEnabled()) {
+    if (EnvUtils.getPropertyAsBool(EXITABLE_READER_PROPERTY) || !queryLimits.isLimitsEnabled()) {
```

Comment:
should this instead check if the existing reader is an instance of ExitableDirectoryReader?  It would have already been wrapped.